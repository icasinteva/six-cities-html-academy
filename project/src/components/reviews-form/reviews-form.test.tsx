import { configureMockStore } from '@jedmao/redux-mock-store';
import { render } from '@testing-library/react';

import { Provider } from 'react-redux';
import ReviewsForm from './reviews-form';


const mockStore = configureMockStore();
const store = mockStore({
  REVIEWS_FORM: {},
});
const onSubmit = jest.fn();

describe('Component: ReviewsForm', () => {
  it('should render correctly', () => {
    const { container } = render(
      <Provider store={store}>
        <ReviewsForm hotelId='17' onSubmit={onSubmit} />
      </Provider>,
    );

    expect(container).toMatchSnapshot();
  });
});
