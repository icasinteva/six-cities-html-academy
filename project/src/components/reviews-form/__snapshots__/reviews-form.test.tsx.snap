// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: ReviewsForm should render correctly 1`] = `
<div>
  <form
    action=""
    class="reviews__form form"
    data-testid="reviews__form"
  >
    <label
      class="reviews__label form__label"
      for="review"
    >
      Your review
    </label>
    <div
      class="reviews__rating-form form__rating"
    >
      <input
        class="form__rating-input visually-hidden"
        data-testid="5-stars"
        id="5-stars"
        name="rating"
        type="radio"
        value="5"
      />
      <label
        class="reviews__rating-label form__rating-label"
        for="5-stars"
        title="perfect"
      >
        <svg
          class="form__star-image"
          height="33"
          width="37"
        >
          <use
            xlink:href="#icon-star"
          />
        </svg>
      </label>
      <input
        class="form__rating-input visually-hidden"
        data-testid="4-stars"
        id="4-stars"
        name="rating"
        type="radio"
        value="4"
      />
      <label
        class="reviews__rating-label form__rating-label"
        for="4-stars"
        title="good"
      >
        <svg
          class="form__star-image"
          height="33"
          width="37"
        >
          <use
            xlink:href="#icon-star"
          />
        </svg>
      </label>
      <input
        class="form__rating-input visually-hidden"
        data-testid="3-stars"
        id="3-stars"
        name="rating"
        type="radio"
        value="3"
      />
      <label
        class="reviews__rating-label form__rating-label"
        for="3-stars"
        title="not bad"
      >
        <svg
          class="form__star-image"
          height="33"
          width="37"
        >
          <use
            xlink:href="#icon-star"
          />
        </svg>
      </label>
      <input
        class="form__rating-input visually-hidden"
        data-testid="2-stars"
        id="2-stars"
        name="rating"
        type="radio"
        value="2"
      />
      <label
        class="reviews__rating-label form__rating-label"
        for="2-stars"
        title="badly"
      >
        <svg
          class="form__star-image"
          height="33"
          width="37"
        >
          <use
            xlink:href="#icon-star"
          />
        </svg>
      </label>
      <input
        class="form__rating-input visually-hidden"
        data-testid="1-stars"
        id="1-stars"
        name="rating"
        type="radio"
        value="1"
      />
      <label
        class="reviews__rating-label form__rating-label"
        for="1-stars"
        title="terribly"
      >
        <svg
          class="form__star-image"
          height="33"
          width="37"
        >
          <use
            xlink:href="#icon-star"
          />
        </svg>
      </label>
    </div>
    <textarea
      class="reviews__textarea form__textarea"
      data-testid="review"
      id="review"
      maxlength="300"
      minlength="50"
      name="comment"
      placeholder="Tell how was your stay, what you like and what can be improved"
      required=""
    />
    <div
      class="reviews__button-wrapper"
    >
      <p
        class="reviews__help"
      >
        To submit review please make sure to set 
        <span
          class="reviews__star"
        >
          rating
        </span>
         and describe your stay with at least 
        <b
          class="reviews__text-amount"
        >
          50
           characters
        </b>
        .
      </p>
      <input
        class="reviews__submit form__submit button"
        disabled=""
        type="submit"
        value="Submit"
      />
    </div>
  </form>
</div>
`;
